Mlem/App/Views/Root/Tabs/Feeds/SubscriptionListView.swift

```diff
@@ -42,32 +36,39 @@ struct SubscriptionListView: View {
         }
     }
     
+    var subscriptions: SubscriptionList? {
+        (appState.firstSession as? UserSession)?.subscriptions
+    }
+    
+    var sectionIndicesShown: Bool {
+        !UIDevice.isPad && sort == .alphabetical && (subscriptions?.communities.count ?? 0) > 10
+    }
+    
+    var body: some View {
+        Group {
+            // TODO: iOS 18 deprecation remove compatibility shim
+            if #available(iOS 26, *) {
+                content
+                    .listSectionIndexVisibility(sectionIndicesShown ? .visible : .hidden)
+            } else {
+                content
+            }
+        }
+        .listStyle(.sidebar)
+        .navigationTitle("Feeds")
+    }
+    
     @ViewBuilder
     var content: some View {
         let sections = subscriptions?.visibleSections(sort: sort) ?? []
         
         Form {
-            Section {
-                ForEach(feedOptions, id: \.hashValue) { feedOption in
-                    SubscriptionListNavigationButton(.feeds(feedOption)) {
-                        HStack(spacing: 15) {
-                            FeedIconView(
-                                feedDescription: feedOption.description,
-                                size: appState.firstSession is GuestSession ? 36 : 28
-                            )
-                            VStack(alignment: .leading) {
-                                Text(feedOption.description.label)
-                                if appState.firstSession is GuestSession {
-                                    Text(feedOption.description.subtitle)
-                                        .font(.footnote)
-                                        .foregroundStyle(.themedSecondary)
-                                }
-                            }
-                        }
-                        .frame(maxWidth: .infinity, alignment: .leading)
-                        .contentShape(.rect)
-                    }
-                }
+            // TODO: iOS 18 deprecation remove compatibility shim
+            if #available(iOS 26, *) {
+                feeds
+                    .sectionIndexLabel("★")
```

Comment:
There's a version of `.sectionIndexLabel` that accepts a `Text` argument - it may be more idiomatic to write:

```suggestion
                    .sectionIndexLabel(Text(Image(icon: .lemmy.favorite)))
```

We might have to apply `.symbolVariant(.fill)` at the top level, unless SwiftUI applies that automatically. If that can't be made to work, we can use:

```suggestion
                    .sectionIndexLabel(Text(Image(systemName: "star.fill")))
```

Reply:
I can't get this to work--it just doesn't render the image. [This ancient thread](https://stackoverflow.com/questions/5898836/can-use-image-for-sectionindextitlesfortableview) suggests that perhaps it's a limitation of the API.

Reply:
Ok, happy to leave as-is then 👍